### PR TITLE
Added support to push console log to fpaste.org and validate production template

### PR DESCRIPTION
**Pull request by jkandasa**
* Added support to validate [production template](https://github.com/jaegertracing/jaeger-openshift/blob/master/production/jaeger-production-template.yml)
* Added support to publish Jenkins CI server PR job console log to the world via [Fedora pastebin](https://paste.fedoraproject.org/)

### CODE REVIEW

**Comment by objectiser** [comment]
:+1: :white_check_mark: :honeybee: Jenkins CI Build`#1`: `Console log:` [external](https://paste.fedoraproject.org/paste/nR~~TiOhVjFGxj4~vxyKOQ), [internal](https://jenkins-jaeger-ci.cloud2.jonqe.lab.eng.bos.redhat.com/job/jaegertracing/job/jaeger-openshift/job/PR-32/1/display/redirect)

**Comment by pavolloffay** [comment]
What do these emojis mean? Could we use just :red_circle: for fail and :white_check_mark: for success?

I am not sure if `jq` binary should be in this repo.

**Comment by jkandasa** [comment]
@pavolloffay 

>What do these emojis mean? Could we use just :red_circle: for fail and :white_check_mark: for success?

Sure, I will update.

> I am not sure if jq binary should be in this repo.

Please give a suggestion

**Comment by objectiser** [comment]
:+1: :white_check_mark: :honeybee: Jenkins CI Build`#2`: `Console log:` [external](https://paste.fedoraproject.org/paste/-0tEERixZ0WyppmE7ojomg), [internal](https://jenkins-jaeger-ci.cloud2.jonqe.lab.eng.bos.redhat.com/job/jaegertracing/job/jaeger-openshift/job/PR-32/2/display/redirect)

**Comment by objectiser** [comment]
:white_check_mark: Jenkins CI Build`#3`: `Console log:` [external](https://paste.fedoraproject.org/paste/ZuKaIXQl69v5RJ08AYqq9w), [internal](https://jenkins-jaeger-ci.cloud2.jonqe.lab.eng.bos.redhat.com/job/jaegertracing/job/jaeger-openshift/job/PR-32/3/display/redirect)

**Comment by pavolloffay** [comment]
> Please give a suggestion

I think there are two options, directly install it on the machine or have a script here which would install it. I think first is better.

**Comment by objectiser** [comment]
:white_check_mark: Jenkins CI Build`#4`: `Console log:` [external](), [internal](https://jenkins-jaeger-ci.cloud2.jonqe.lab.eng.bos.redhat.com/job/jaegertracing/job/jaeger-openshift/job/PR-32/4/display/redirect)

**Comment by objectiser** [comment]
:white_check_mark: Jenkins CI Build`#5`: `Console log:` [external](), [internal](https://jenkins-jaeger-ci.cloud2.jonqe.lab.eng.bos.redhat.com/job/jaegertracing/job/jaeger-openshift/job/PR-32/5/display/redirect)

**Comment by objectiser** [comment]
:white_check_mark: Jenkins CI Build`#6`: `Console log:` [external](https://paste.fedoraproject.org/paste/k3CUp6GxRxlzBzBQYfISng), [internal](https://jenkins-jaeger-ci.cloud2.jonqe.lab.eng.bos.redhat.com/job/jaegertracing/job/jaeger-openshift/job/PR-32/6/display/redirect)

**Comment by jkandasa** [comment]
@pavolloffay 
>I think there are two options, directly install it on the machine or have a script here which would install it. I think first is better.

I have implemented the second option with a smart way. First check `jq` file availability on `/tmp` location, if it is not available to do download on `/tmp/`. From second time download will not happen.

I am sorry to pollute with too many builds. I forget to delete old `jq` soft link from the workspace. fixed it.

**Comment by objectiser** [comment]
:white_check_mark: Jenkins CI Build`#1`: `Console log:` [external](https://paste.fedoraproject.org/paste/kpX1Jf2Xy-B78J9qB5QaiA), [internal](https://jenkins-jaeger-ci.cloud2.jonqe.lab.eng.bos.redhat.com/job/jaegertracing/job/jaeger-openshift/job/PR-32/1/display/redirect)

File: jenkins-ci/push-logs.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  elif [ "${JOB_STATUS}" = "SUCCESS" ]; then
+    BUILD_STATUS_ICON=":white_check_mark:"
+  else
+    BUILD_STATUS_ICON=":question:"
+  fi
+  echo "Console log on fpaste: ${FPASTE_URL}"
+  curl -H "Content-Type: application/json" \
+    -u $AUTHTOKEN \
+    -X POST \
+    -d "{\"body\":\"${BUILD_STATUS_ICON} Jenkins CI Build\`#${BUILD_ID}\`: \`Console log:\` [external](${FPASTE_URL}), [internal](${RUN_DISPLAY_URL})\"}" \
```
Comment (pavolloffay) [inline jenkins-ci/push-logs.sh]: \`Console log:\` can be just Console log

File: jenkins-ci/push-logs.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    echo "jq file not found! downloading jq to '/tmp/jq-linux64'"
+    curl -sL https://github.com/stedolan/jq/releases/download/jq-1.5/jq-linux64 > /tmp/jq-linux64
+    chmod +x /tmp/jq-linux64
+  fi
+  if [ ! -e jq ]; then
+    echo "jq file 'soft link' not found! creating it on '$(pwd)/jq'"
+    ln -s /tmp/jq-linux64 jq
+  fi
+  ./jq --version
+  #consoleUrl=$(echo "http://localhost:8080/job/${JOB_NAME}/${BUILD_ID}/consoleText")
```
Comment (pavolloffay) [inline jenkins-ci/push-logs.sh]: Remove if not needed

**Comment by objectiser** [comment]
:white_check_mark: Jenkins CI Build`#2`: `Console log:` [external](https://paste.fedoraproject.org/paste/UEthvzY-Ks7SUTVjvp5F9w), [internal](https://jenkins-jaeger-ci.cloud2.jonqe.lab.eng.bos.redhat.com/job/jaegertracing/job/jaeger-openshift/job/PR-32/2/display/redirect)

**Comment by objectiser** [comment]
:white_check_mark: Jenkins CI Build`#3`: `Console log:` [external](https://paste.fedoraproject.org/paste/q57gZajG8-gKSt80sr9D1A), [internal](https://jenkins-jaeger-ci.cloud2.jonqe.lab.eng.bos.redhat.com/job/jaegertracing/job/jaeger-openshift/job/PR-32/3/display/redirect)

**Comment by objectiser** [comment]
:white_check_mark: Jenkins CI Build`#4`: Console log: [external](https://paste.fedoraproject.org/paste/umOgwTUGxXzWXj5XY6~wFg), [internal](https://jenkins-jaeger-ci.cloud2.jonqe.lab.eng.bos.redhat.com/job/jaegertracing/job/jaeger-openshift/job/PR-32/4/display/redirect)

File: jenkins-ci/push-logs.sh (hunk 1 above)
Comment (jkandasa) [inline jenkins-ci/push-logs.sh]: @pavolloffay removed `

**Comment by pavolloffay** [comment]
I think it could go in. 

cc @objectiser @jpkrohling

**Comment by objectiser** [comment]
LGTM 👍

**Comment by pavolloffay** [comment]
@jkandasa thanks!

**Pull request merged**